pkg/transaction/monitor.go

```diff
@@ -169,44 +171,34 @@ func (tm *transactionMonitor) watchPending() {
 	}
 }
 
-// potentiallyConfirmedTxWatches returns all watches with nonce less than what was specified
-func (tm *transactionMonitor) potentiallyConfirmedTxWatches(nonce uint64) (watches map[uint64]map[common.Hash][]transactionWatch) {
+func (tm *transactionMonitor) hasWatches() bool {
 	tm.lock.Lock()
 	defer tm.lock.Unlock()
+	return len(tm.watchesByNonce) > 0
+}
 
-	potentiallyConfirmedTxWatches := make(map[uint64]map[common.Hash][]transactionWatch)
-	for n, watches := range tm.watchesByNonce {
-		if n < nonce {
-			potentiallyConfirmedTxWatches[n] = watches
+func watchStart(watches []transactionWatch) time.Time {
+	var start time.Time
+	for _, w := range watches {
+		if w.start.Before(start) {
+			start = w.start
 		}
 	}
-
-	return potentiallyConfirmedTxWatches
+	return start
 }
 
-func (tm *transactionMonitor) hasWatches() bool {
-	tm.lock.Lock()
-	defer tm.lock.Unlock()
-	return len(tm.watchesByNonce) > 0
-}
+// Time for transaction is checked if it is accepted for mining.
+const transactionWatchNotFoundTimeout = 20 * time.Second
```

Comment:
how is this defined at 20 sec?
maybe this should be a multiple of the block time of the blockchain?

Reply:
This is a completely rough estimate. A block time multiple is a very good approach.